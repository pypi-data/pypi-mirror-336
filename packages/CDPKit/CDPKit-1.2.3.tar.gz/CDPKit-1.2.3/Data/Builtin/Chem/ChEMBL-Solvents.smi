WATER	[OH2]
DICHLOROMETHANE	ClCCl
TRICHLOROMETHANE	ClC(Cl)Cl
ETHYL ACETATE	CCOC(=O)C
METHANOL	CO
PROPAN-2-OL	CC(C)O
ACETONE	CC(=O)C
DMSO	CS(=O)C
ETHANOL	CCO
