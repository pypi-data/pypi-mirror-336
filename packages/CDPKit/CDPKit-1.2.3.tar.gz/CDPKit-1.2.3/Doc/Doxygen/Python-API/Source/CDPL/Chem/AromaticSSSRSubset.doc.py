#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief Implements the extraction of the aromatic rings in the SSSR of a molecular graph.
# 
class AromaticSSSRSubset(FragmentList):

    ##
    # \brief Constructs an empty <tt>AromaticSSSRSubset</tt> instance.
    # 
    def __init__() -> None: pass

    ##
    # \brief Initializes a copy of the \e %AromaticSSSRSubset instance \a arom_sssr.
    # \param arom_sssr The \e %AromaticSSSRSubset instance to copy.
    # 
    def __init__(arom_sssr: AromaticSSSRSubset) -> None: pass

    ##
    # \brief Construct a <tt>AromaticSSSRSubset</tt> instance that contains all aromatic rings of the SSSR of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to extract the aromatic rings.
    # 
    def __init__(molgraph: MolecularGraph) -> None: pass

    ##
    # \brief Replaces the current state of \a self with a copy of the state of the \c %AromaticSSSRSubset instance \a arom_sssr.
    # \param arom_sssr The \c %AromaticSSSRSubset instance to copy.
    # \return \a self
    # 
    def assign(arom_sssr: AromaticSSSRSubset) -> AromaticSSSRSubset: pass

    ##
    # \brief Replaces the current set of rings by the aromatic rings in the SSSR of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to extract the aromatic rings.
    # 
    def extract(molgraph: MolecularGraph) -> None: pass
