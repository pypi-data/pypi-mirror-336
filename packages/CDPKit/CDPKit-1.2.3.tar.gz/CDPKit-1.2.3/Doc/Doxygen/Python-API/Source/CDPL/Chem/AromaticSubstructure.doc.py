#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief Implements the perception of aromatic atoms and bonds in a molecular graph.
# 
class AromaticSubstructure(Fragment):

    ##
    # \brief Constructs an empty <tt>AromaticSubstructure</tt> instance.
    # 
    def __init__() -> None: pass

    ##
    # \brief Construct a <tt>AromaticSubstructure</tt> instance that consists of the aromatic atoms and bonds of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to perceive aromatic atoms and bonds.
    # 
    def __init__(molgraph: MolecularGraph) -> None: pass

    ##
    # \brief Replaces the currently stored atoms and bonds by the set of aromatic atoms and bonds of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to perceive aromatic atoms and bonds.
    # 
    def perceive(molgraph: MolecularGraph) -> None: pass
