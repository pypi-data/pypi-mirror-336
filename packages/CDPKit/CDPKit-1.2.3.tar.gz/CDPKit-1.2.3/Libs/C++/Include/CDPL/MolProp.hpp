/* 
 * MolProp.hpp 
 *
 * This file is part of the Chemical Data Processing Toolkit
 *
 * Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this library; see the file COPYING. If not, write to
 * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA.
 */

/**
 * \file
 * \brief A convenience header including everything that is defined in namespace CDPL::MolProp.
 */

#ifndef CDPL_MOLPROP_HPP
#define CDPL_MOLPROP_HPP

#include "CDPL/MolProp/AtomProperty.hpp"
#include "CDPL/MolProp/BondProperty.hpp"
#include "CDPL/MolProp/AtomPropertyDefault.hpp"
#include "CDPL/MolProp/MassComposition.hpp"
#include "CDPL/MolProp/ElementHistogram.hpp"
#include "CDPL/MolProp/CoordinationGeometry.hpp"
#include "CDPL/MolProp/HBondDonorAtomType.hpp"
#include "CDPL/MolProp/HBondAcceptorAtomType.hpp"
#include "CDPL/MolProp/AtomFunctions.hpp"
#include "CDPL/MolProp/BondFunctions.hpp"
#include "CDPL/MolProp/AtomContainerFunctions.hpp"
#include "CDPL/MolProp/BondContainerFunctions.hpp"
#include "CDPL/MolProp/MolecularGraphFunctions.hpp"
#include "CDPL/MolProp/TPSACalculator.hpp"
#include "CDPL/MolProp/AtomHydrophobicityCalculator.hpp"
#include "CDPL/MolProp/LogSCalculator.hpp"
#include "CDPL/MolProp/XLogPCalculator.hpp"
#include "CDPL/MolProp/PEOESigmaChargeCalculator.hpp"
#include "CDPL/MolProp/MHMOPiChargeCalculator.hpp"
#include "CDPL/MolProp/HBondDonorAtomTyper.hpp"
#include "CDPL/MolProp/HBondAcceptorAtomTyper.hpp"

#endif // CDPL_MOLPROP_HPP
