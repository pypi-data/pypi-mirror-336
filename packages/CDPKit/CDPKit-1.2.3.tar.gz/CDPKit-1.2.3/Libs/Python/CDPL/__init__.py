##
# __init__.py  
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
##


from __future__ import absolute_import

import os
import sys
import platform

if sys.version_info[1] >= 8 and platform.system() == 'Windows':
    if 'site-packages' in __file__:
        os.add_dll_directory(os.path.dirname(__file__))
    else:
        os.add_dll_directory(os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(__file__))), '@CDPKIT_EXECUTABLE_INSTALL_DIR@'))

from ._cdpl import *

__version__ = VERSION_STRING
