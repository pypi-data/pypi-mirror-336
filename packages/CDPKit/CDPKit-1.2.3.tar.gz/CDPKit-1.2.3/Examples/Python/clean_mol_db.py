#!/bin/env python

##
# clean_mol_db.py 
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# Permission to use, copy, modify, and/or distribute this software for any purpose with
# or without fee is hereby granted.
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD
# TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN
# NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
# DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
# IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
##


import sys
import argparse

import CDPL.Chem as Chem
import CDPL.MolProp as MolProp

    
# performs a (optional) standardization of the argument molecule and then checks whether
# it fulfills certain user-defined criteria for inclusion/exclusion
def processMolecule(mol: Chem.Molecule, args: argparse.Namespace) -> tuple:
    chgs_mod = False

    if args.min_charges:
        # will store the 'uncharged' version of the argument mol
        res_mol = Chem.BasicMolecule() 

        # minimize the number of charged atoms using the corresponding functionality implemented
        # in class Chem.ProtonationStateStandardizer
        chgs_mod = Chem.ProtonationStateStandardizer().standardize(mol, res_mol, Chem.ProtonationStateStandardizer.MIN_CHARGED_ATOM_COUNT)

        # if changes were made then from now on use the 'uncharged' molecule for further checks
        if chgs_mod:  
            mol = res_mol
        
    comps_strpd = False
    
    if args.strip_comps:
        # perceive components (if not done already)
        comps = Chem.perceiveComponents(mol, False) 

        # find largest component (regarding heavy atom count) but only if multi-comp. molecule
        if comps.size > 1:    # check if multi-comp. molecule
            lgst_comp = None
            lgst_comp_hvy_atom_count = 0

            for comp in Chem.getComponents(mol):                 # for each component
                hvy_atom_count = MolProp.getHeavyAtomCount(comp) # calc. non-hydrogen atom count

                if hvy_atom_count > lgst_comp_hvy_atom_count:    # check if the largest comp. so far has been found
                    lgst_comp_hvy_atom_count = hvy_atom_count    # if so, store for later use
                    lgst_comp = comp

            # if the input molecule has structure data then pass them on (for SDF output)
            if Chem.hasStructureData(mol):                       
                Chem.setStructureData(lgst_comp, Chem.getStructureData(mol))

            # for further checks use only the identified largest component 
            comps_strpd = True
            mol = lgst_comp

    # calc. implicit hydrogen counts (if not done already)
    Chem.calcImplicitHydrogenCounts(mol, False)
    
    # create instance of class MolProp.ElementHistogram for storing the per-element atom counts
    # of the molecule (or its largest comp.)
    elem_histo = MolProp.ElementHistogram()              

    # get per-element atom counts
    MolProp.generateElementHistogram(mol, elem_histo, False)    

    # check if the found chem. elements are all in the set of allowed elements (if specified)
    if not checkAllowedElements(elem_histo, args.allowed_elements):    
        return (None, 'prohibited element detected')

    # check if none of the found chem. elements is in the set of excluded elements (if specified)
    if not checkExcludedElements(elem_histo, args.excluded_elements):
        return (None, 'prohibited element detected')

    # check if all specified minium chem. element atom counts are reached
    if not checkMinAtomCounts(elem_histo, args.min_atom_counts):
        return (None, 'minimum atom counts not reached')

    # check if none of the specified maximum chem. element atom counts gets exceeded
    if not checkMaxAtomCounts(elem_histo, args.max_atom_counts):
        return (None, 'maximum atom count exceeded')

    log_msg = None
    
    if chgs_mod:
        log_msg = 'reduced charged atom count'

    if comps_strpd:
        if log_msg:
            log_msg += ', removed components'
        else:
            log_msg = 'removed components'
            
    return (mol, log_msg)

# checks if all found chem. elements are in the set of allowed elements (if specified)
def checkAllowedElements(elem_histo: MolProp.ElementHistogram, elem_list: list) -> bool:
    if not elem_list:
        return True

    for atom_type in elem_histo.getKeys():
        allowed = False
        
        for all_atom_type in elem_list:
            if Chem.atomTypesMatch(all_atom_type, atom_type):
                allowed = True
                break

        if not allowed:
            return False

    return True

# checks if none of the found chem. elements is in the set of excluded elements (if specified)
def checkExcludedElements(elem_histo: MolProp.ElementHistogram, elem_list: list) -> bool:
    if not elem_list:
        return True

    for atom_type in elem_histo.getKeys():
        for x_atom_type in elem_list:
            if Chem.atomTypesMatch(x_atom_type, atom_type):
                return False

    return True

# return the total number of atoms matching the specified atom type (element or generic type)
def getNumAtomsOfType(elem_histo: MolProp.ElementHistogram, qry_atom_type: int) -> int:
    tot_count = 0

    for atom_type, count in elem_histo.getEntries():
        # check if the elem. histogram entry matches the specified atom type (element or generic type)
        # if so, add the stored count the total count
        if Chem.atomTypesMatch(qry_atom_type, atom_type):  
            tot_count += count

    return tot_count

# checks if all the specified minium counts of atoms matching a particular element or
# generic type are reached
def checkMinAtomCounts(elem_histo: MolProp.ElementHistogram, atom_counts: dict) -> bool:
    if not atom_counts:
        return True
    
    for atom_type, min_count in atom_counts.items():
        if getNumAtomsOfType(elem_histo, atom_type) < min_count:
            return False

    return True

# checks if none of the specified maximum counts of atoms matching a particular element or
# generic type gets exceeded
def checkMaxAtomCounts(elem_histo: MolProp.ElementHistogram, atom_counts: dict) -> bool:
    if not atom_counts:
        return True
    
    for atom_type, max_count in atom_counts.items():
        if getNumAtomsOfType(elem_histo, atom_type) > max_count:
            return False

    return True

# converts a comma separated list of chem. element/generic type symbols into a list of the
# corresponding numeric atom types defined in Chem.AtomType
def parseElementList(elem_list: str) -> list:
    if not elem_list:
        return []
    
    atom_types = []
    
    for elem in elem_list.split(','):
        elem = elem.strip()
        
        if not elem: # zero length?
            continue
        
        atom_type = Chem.AtomDictionary.getType(elem)

        if atom_type == Chem.AtomType.UNKNOWN:
            sys.exit('Error: unknown chemical element/generic type \'%s\'' % elem)

        atom_types.append(atom_type)
            
    return atom_types

# converts a comma separated list of chem. element (or generic type) symbol/count pairs (sep. by colon) into a dictionary
# mapping the corresponding numeric atom types (defined in Chem.AtomType) to an integer number
def parseElementCountList(elem_count_list: str) -> dict:
    if not elem_count_list:
        return {}
    
    atom_type_counts = {}
    
    for elem_count in elem_count_list.split(','):
        elem_count = elem_count.strip()
        
        if not elem_count: # zero length?
            continue
        
        elem_count = elem_count.split(':')
        elem_count[0] = elem_count[0].strip()
        atom_type = Chem.AtomDictionary.getType(elem_count[0])
        
        if atom_type == Chem.AtomType.UNKNOWN:
            sys.exit('Error: unknown chemical element/generic type \'%s\'' % elem_count[0])

        count = 1
        
        if len(elem_count) > 1: # has count spec.?
            count = int(elem_count[1])
            
        atom_type_counts[atom_type] = count
            
    return atom_type_counts
                                
def parseArgs() -> argparse.Namespace:
    parser = argparse.ArgumentParser(description='Strips compounds that fulfill particular user-defined criteria from a molecule database.')

    parser.add_argument('-i',
                        dest='in_file',
                        required=True,
                        metavar='<file>',
                        help='Input molecule file')
    parser.add_argument('-o',
                        dest='out_file',
                        required=True,
                        metavar='<file>',
                        help='Output molecule file')
    parser.add_argument('-d',
                        dest='disc_file',
                        required=False,
                        metavar='<file>',
                        help='Discarded molecule output file')
    parser.add_argument('-s',
                        dest='strip_comps',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Keep only the largest molecule component (default: false)')
    parser.add_argument('-c',
                        dest='min_charges',
                        required=False,
                        action='store_true',
                        default=False,
                        help='Minimize number of charged atoms (default: false)')
    parser.add_argument('-x',
                        dest='excluded_elements',
                        required=False,
                        metavar="<element list>",
                        help='List of excluded chem. elements (default: no elements are excluded)')
    parser.add_argument('-a',
                        dest='allowed_elements',
                        required=False,
                        metavar="<element list>",
                        help='List of allowed chem. elements (default: all elements are allowed)')
    parser.add_argument('-m',
                        dest='min_atom_counts',
                        required=False,
                        metavar="<element count list>",
                        help='Minimum chem. element specific atom counts (default: no count limits)')
    parser.add_argument('-M',
                        dest='max_atom_counts',
                        required=False,
                        metavar="<element count list>",
                        help='Maximum chem. element specific atom counts (default: no count limits)')
    parser.add_argument('-v',
                        dest='verb_level',
                        required=False,
                        metavar='<0|1|2|3>',
                        choices=range(0, 4),
                        default=1,
                        help='Verbosity level (default: 1; 0 -> no console output, 1 -> print summary, 2 -> verbose, 3 -> extra verbose)',
                        type=int)

    return parser.parse_args()

def main() -> None:
    args = parseArgs() # process command line arguments

    # convert specified lists of chem. element/gen. type symbols into a corresponding list of
    # numeric atom types (defined in Chem.Atomtype)
    args.allowed_elements  = parseElementList(args.allowed_elements)
    args.excluded_elements = parseElementList(args.excluded_elements)

    # convert specified comma separated lists of chem. element (or generic type) symbol/count pairs (sep. by colon) into
    # corresponding dictionaries mapping numeric atom types (defined in Chem.AtomType) to integers
    args.min_atom_counts = parseElementCountList(args.min_atom_counts)
    args.max_atom_counts = parseElementCountList(args.max_atom_counts)
    
    # create reader for input molecules (format specified by file extension)
    reader = Chem.MoleculeReader(args.in_file) 

    # create writer for molecules passing the checks (format specified by file extension)
    writer = Chem.MolecularGraphWriter(args.out_file) 

    # write canonical SMILES
    Chem.setSMILESOutputCanonicalFormParameter(writer, True)
    
    if args.disc_file:
        # create writer for sorted out molecules (format specified by file extension)
        disc_writer = Chem.MolecularGraphWriter(args.disc_file)
        # write canonical SMILES
        Chem.setSMILESOutputCanonicalFormParameter(disc_writer, True)
    else:
        disc_writer = None
        
    # create instances of the default implementation of the Chem.Molecule interface for the input and output molecules
    in_mol = Chem.BasicMolecule()

    i = 0
    num_changed = 0
    num_disc = 0
    
    try:
        # read and process molecules one after the other until the end of input has been reached (or a severe error occurs)
        while reader.read(in_mol):
            # compose a molecule identifier
            mol_id = Chem.getName(in_mol).strip() 

            if mol_id == '':
                mol_id = '#' + str(i + 1)  # fallback if name is empty or not available
            else:
                mol_id = '\'%s\' (#%s)' % (mol_id, str(i + 1))
            
            try:
                # process input molecule
                out_mol, log_msg = processMolecule(in_mol, args) 
                
                if not out_mol:         # check whether the molecule has been sorted out
                    if args.verb_level > 1 and log_msg:
                        print('- Molecule %s: discarded, %s' % (mol_id, log_msg))

                    num_disc += 1
                        
                    if not disc_writer: # check whether discarded molecules should be saved to a separate file
                        i += 1
                        continue

                    # write discarded molecule to the specified target file
                    out_mol = in_mol
                    out_mol_writer = disc_writer
                else:
                    if log_msg:
                        if args.verb_level > 1 :
                            print('- Molecule %s: %s' % (mol_id, log_msg))
                            
                        num_changed += 1
                    else:
                        if args.verb_level > 2:
                            print('- Molecule %s: left unchanged' % mol_id)

                    # molecule passed all checks and needs to be written to the regular output file
                    out_mol_writer = writer
                    
                try:
                    # calculate (if not already present) some basic properties of the output molecule
                    # that might be required for writing (output format dependent)
                    Chem.calcImplicitHydrogenCounts(out_mol, False)
                    Chem.perceiveHybridizationStates(out_mol, False)
                    Chem.perceiveSSSR(out_mol, False)
                    Chem.setRingFlags(out_mol, False)
                    Chem.setAromaticityFlags(out_mol, False)
                    Chem.perceiveComponents(out_mol, False)
                              
                    # output molecule
                    if not out_mol_writer.write(out_mol):
                        sys.exit('Error: writing molecule %s failed' % mol_id)

                except Exception as e: # handle exception raised in case of severe write errors
                    sys.exit('Error: writing molecule %s failed: %s' % (mol_id, str(e)))

                i += 1
                
            except Exception as e: # handle exception raised in case of severe processing errors
                sys.exit('Error: processing of molecule %s failed: %s' % (mol_id, str(e)))
                
    except Exception as e: # handle exception raised in case of severe read errors
        sys.exit('Error: reading of molecule %s failed: %s' % (str(i), str(e)))

    if args.verb_level > 0:
        print('Summary:')
        print(' - %s molecules processed' % str(i))
        print(' - %s modified ' % str(num_changed))
        print(' - %s discarded' % str(num_disc))
        
    writer.close()
    sys.exit(0)

if __name__ == '__main__':
    main()
