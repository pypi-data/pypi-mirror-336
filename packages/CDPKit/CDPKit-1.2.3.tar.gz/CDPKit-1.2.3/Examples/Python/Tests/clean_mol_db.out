c1(c(c(c2cnc(cn2)N)ccc1C1CCC1)F)Oc1cc(nc(Cc2ccccc2)n1)N
C1(C(c2c(cc(cc2)C)NC1=O)=O)=NNc1c(cccc1)Cl
C([C@H]([C@H](CC)C)Nc1c2c3c(CCCC3)sc2nc(C)n1)(=O)O
C([C@@]1(c2ccccc2)[C@@H](C1)CN)(N(CC)CC)=O
C([C@@]1(c2ccccc2)[C@@H](C1)CN)(N(CC)CC)=O
C(C1C=C(C)OC(C=1)=O)#CCCCC
C1#CC[C@H]([C@H](CC#CC=C1)CO)CO
C(=CCN([C@@H](c1ccccc1)C)C)=CC
C(=CCN([C@@H](c1ccccc1)C)C)=CC
C([C@H](CCC(Nc1ccc(cc1)[N](=O)=O)=O)N)(=O)O
c1(c(c(cnc1C[S@](c1nc2c(ccc(c2)OC)[nH]1)=O)C)OC)C
C(C1C(C(C(C(O1)OC1C(C(OC2C(C(=O)O)OC(C(C2O)O)O)OC(C1O)CO)NC(C)=O)O)O)OC1C(C(C(C(CO)O1)O)O)NC(C)=O)(NCCOC(Cc1c(cccc1)Nc1c(cccc1Cl)Cl)=O)=O
C12C(=CC(C(=C1)Nc1ccc(cc1)Cl)=NC(C)C)N(c1c(cccc1)N=2)c2ccc(cc2)Cl
C(c1ccc(cc1)N(C)C)(CC(C[C@H]1[C@@H]([C@@H]([C@@H](C[C@H]2[C@H]([C@H]([C@H](C)O)C)O2)CO1)O)O)=O)=O
C(c1c(cc(cc1F)C)F)(N[C@@H]1c2nc(c3c(cc(cc3)NC(OC)=O)NC(CCC=CC1)=O)c([nH]2)Cl)=O  
C1(COC(=N1)Nc1cc2c(cc1)ncnc2Nc2cc(c(cc2)Oc2cc1ncnn1cc2)C)(C)C
C(c1c(ccc(c1)[N+]([O-])=O)OC(c1ccc(cc1)NC)=O)#N
C12[C@@]34[C@@]5([C@@H]6[C@@]7([C@@H]([C@@](CC[C@@H]7OC)(CN6CC)C)[C@@H]5O)[C@@]1(C[C@@H]([C@@H]2OC)[C@H](C3)OC)O)OCO4
C12C(C3CCC1N3C(c3ccc1c(ccc(c1)Oc1ccc(C(F)(F)F)cn1)n3)=O)CNC2 BDBM250863
C([C@@]1([C@H](C2CC2)C1)NC([C@@H]1C[C@H](CN1C([C@H](C(C)(C)C)NC(O[C@@H]1C[C@H]2[C@H](C2)C1)=O)=O)Oc1c2c(c(cn1)OC)ccc(c2)Cl)=O)(NS(C2CC2)(=O)=O)=O  
c1([C@H]2C[C@H]3CC[C@H](C2)N3CCCOc3ccccc3)cc(ccc1)O  
c12c(nc(Cc3ccccc3)nc1[O+]([CH-]C=C2)CC)NCc2ccccc2 BDBM163075
C1(N(c2ccc(cc2)N2CCN(c3ccc(cc3)OC[C@H]3CO[C@@](c4c(cc(cc4)Cl)Cl)(Cn4cncn4)O3)CC2)C=NN1C(CC)C)=O
C(c1cc(N2CCCC2)ncc1)(Nc1cc(c(cc1)C)c1ccc(C(NCC2CC2)=O)cc1)=O
c12c([N]([C@H]3[C@@H]([C@@H]([C@@H](COP(=O)(O)O)O3)O)O)=CN1)ncnc2S
c1(c(ccc([C@H](CNC)O)c1)O)O
C1(C(Cc2c(cc(cc2)[N+]([O-])=O)C)(C)O)C(N(c2ccccc2)NC=1N)=O
c1(c(c(cnc1CS(c1nc2c(ccc(c2)OC)[nH]1)=O)C)OC)C  
c12c3n([C@H]4[C@@H]([C@@H]([C@@H](COP(OP(OC[C@@H]5[C@H]([C@H]([C@@H](C5)[n+](c1N)cn3)O)O)(=O)O)(=O)O)O4)O)O)cn2
C1(c2ccccc2)(C[N+](CCO1)(C[18F])C)O
C1(C(=O)O)=C(CS[C@@H]2[C@@H](C(N12)=O)NC(C(c2nc(NP(=O)(O)O)sn2)=NOCC)=O)Sc2nc(c1cc[n+](cc1)C)cs2
c12c3n([C@H]4[C@@H]([C@@H]([C@@H](COP(OP(OC[C@@H]5[C@H]([C@H]([C@@H](C5)[n+](c1N)cn3)O)O)(=O)O)(=O)O)O4)O)O)cn2
C(=C[N+]([O-])=O)(NCCSCc1ccc(CN(C)C)o1)NC
C(c1ccc(cc1)N(C)C)(CC(C[C@H]1[C@@H]([C@@H]([C@@H](C[C@H]2[C@H]([C@H]([C@H](C)O)C)O2)CO1)O)O)=O)=O
c12c(cccc1)[n+]1ccc2NCCCCCNc2c3c(cccc3)[n+](cc2)CC=CC1
C(c1c(cccc1)N)(C=Cc1ccccc1)=O
C1(C(C(C(C(CCCC2(C(CC(C(=Cc3csc(C)n3)C)OC(CC1O)=O)O2)C)C)O)C)=O)(C)C 144206291
C1(C(Cc2c(cc(cc2)[N](=O)=O)C)(C)O)C(N(c2ccccc2)NC=1N)=O
c12c(c3c(cccc3)nc1cc(cc2)N=[N]=N)Nc2c(cc(cc2)NS(C)(=O)=O)OC
C1(C(N(C(N2C=1C=CC=C2)=O)CCCCN2CCC(c1c3c(C=CC(F)=[F]3)[nH]c1)CC2)=O)c2ccc(cc2)F
C12C(c3c4c1cccc4ccc3)=[N](c3c(cccc3)N=2)=NC(C(C)Br)=O
C1(C(C)(C)[N+](=C(c2ccc(cc2)OCC(N[C@H](C(=O)O)CCCNC(=N)N)=O)N1O)[O-])(C)C
C(c1cn(C)nc1)(N1[C@@H]2[C@@H](CC1)N(Cc1cccnc1)CC2)=O
C1(C(Nc2c1nccc2)=O)=Cc2c1c(cccc1)n(c2)C
C12C3(c4c(ccc(c4)OC)CC1N(C)CC3)CCCC2
C(CCCCCCCCCCCCCCc1ccc(cc1)[123I])(=O)O
C(c1c(c(c2ccc(cc2)F)n(c1C(C)C)CCC(CC(CC(=O)O)O)O)c1ccccc1)(Nc1ccccc1)=O
[H](C(C)Nc1nc2c(ccc(C(=NC(c3cc(c(cc3)Cl)F)c3cn(C)nc3)O)c2)cn1)CO BDBM192334
C1([C@]23[C@H]([C@@]4([C@]([C@]5([C@@H]([C@@]6([C@H]([C@@]([C@H](CC6)O[C@H]6[C@@H]([C@H]([C@H](CO6)OC(C)=O)OC(C)=O)O[C@H]6[C@@H]([C@@H]([C@H]([C@@H](C)O6)OC(C)=O)OC(C)=O)OC(C)=O)(COC(C)=O)C)CC5)C)CC4O)C)(CC2)C)O1)CC(CC3)(C)C)=O  
c1(ccccc1)CC(C)N
c1(ccccc1)CC(C)N
C1(C(c2c(N(C=1)CC)nc(cc2)C)=O)C(=O)O
C(c1ccc[n+](C2[C@@H]([C@@H](C(COP(=O)(O)O)O2)O)O)c1)(N)=O
C(C(CCCNC(=N)N)[N+](C)(C)C)(OC(C(c1cc(c(cc1)OS(=O)(=O)O)COS(=O)(=O)O)OS(=O)(=O)O)C)=O
C1(c2c(ccc(c2)S(N2CCN(CC2)C)(=O)=O)OCC)=Nc2c(C(N1)=O)n(C)nc2CCC
C1(C(NC(NC=1CN1C(CCC1)=N)=O)=O)Cl
C1(C(C(=C2C(N3[C@@H]4C[C@@H](CC4)O[C@@H]3CN2C=1)=O)O)=O)C(NCc1c(cc(cc1F)F)F)=O
C(CCc1nc(c(c2ccccc2)o1)c1ccccc1)(=O)O
c1(c(c(cnc1CS(c1nc2c(ccc(c2)OC)[nH]1)=O)C)OC)C
c1(c(c(cnc1CS(c1nc2c(ccc(c2)OC)[nH]1)=O)C)OC)C
C(c1c(c(c2cc3c(cc(cc3)NS(C)(=O)=O)cc2)cc(c1)N1C(NC(C=C1)=O)=O)OC)(C)(C)C
C(C(O[C@@H]1C[C@@H]2[C@@H]3[C@H]([C@H](C1)[N+]2(C)C)O3)=O)(c3cccs3)(c3cccs3)O
[2H]C1(c2ccccc2)O[C@H]([C@@H]2C(=C(C(O2)=O)O)O)CO1
[2H]C1(c2ccccc2)O[C@H]([C@@H]2C(=C(C(O2)=O)O)O)CO1
C(c1c(cc(c(c1)Cl)N)OC)(NC1C(CN(CC1)CCCOc1ccc(cc1)F)OC)=O
c1(c(cc(cc1OC)Cc1c(nc(nc1)N)N)OC)OC
C(COc1ccc(cc1)Cl)(NCCN(CC)CC)=O
C1(C(N2C(C(=CC=C2)C)=NC=1)=O)c1nnn[nH]1
C1(C(=CN(C=C1[131I])CC(=O)O)[131I])=O
C1CC1CNC(c1ccc(cc1)c1c(C)ccc(c1)NC(c1cc(N2CCCC2)ncc1)=O)=O
c1(c(cc(cc1Cl)O)Cl)C(C(c1c(cc(cc1Cl)O)Cl)N)N